### PR TITLE
Added checkfile for users with no zone (main)

### PR DESCRIPTION
**Pull request by ganning127**
If they have the needed specific query defined:

Screenshot 2023-06-20 at 1 45 43 PM

Screenshot 2023-06-20 at 1 47 49 PM

They they do not have specific query defined:

Screenshot 2023-06-20 at 1 48 59 PM

Screenshot 2023-06-20 at 1 47 18 PM

### CODE REVIEW

**Comment by trel** [comment]
could lean into the use of https://github.com/irods/irods_client_zone_management_tool/issues/160 once it exists...

**Comment by ganning127** [comment]
Added support for pre-populating specific query and alias
Screenshot 2023-06-29 at 1 24 58 PM

Screenshot 2023-06-29 at 1 25 27 PM

File: src/data/checkfiles/default_checkfiles/validate-user-zone.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ export default {
 
     if (warningAboutSpecificQuery) {
 
-        result.message.push(<span key='push2'>. Additionally, please define the specific query: <Chip label={specificQuery} /> to get faster results.</span>)
+        result.message.push(<span key='push2'>. Additionally, please define the specific query: <Chip label={specificQuery} /> manually, by using this <a rel="noreferrer" target="_blank" href={`/specific-query?sqlStr=${specificQuery}&alias=findUsersWithInvalidZones`}>dynamically generated link</a> to get faster results.</span>)
```
Comment (trel) [inline src/data/checkfiles/default_checkfiles/validate-user-zone.js:132]: "manually, by" -> "manually, or by" ?

**Comment by ganning127** [comment]
had to rebase onto main to test specific query adding

**Comment by trel** [comment]
when you think this is ready - please squash to one, no #.

**Comment by ganning127** [comment]
Coming across a very weird issue...

When I try to add the specific query `SELECT user_name, zone_name FROM R_USER_MAIN WHERE zone_name NOT IN (SELECT zone_name FROM R_ZONE_MAIN)` with the alias `findUsersWithInvalidZones`, I'm getting the error `json.exception.type_error.316` every so often.  

However, when adding that same specific query with another alias, like `asdfasdfasdfasf` or `usersInvalidZones`, I'm never getting that error. 

For reference, when an alias is already taken, the error message is `Error on rcGeneralAdmin`.

**Comment by ganning127** [comment]
force pushed, will take another look at this PR soon before final review

File: src/data/checkfiles/default_checkfiles/validate-user-zone.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import { Link } from "@reach/router";
+import axios from 'axios'
+import Chip from '@material-ui/core/Chip';
+
+export default {
+  name: "Each user is in a valid zone.",
+  description: `Checks if any users belong to invalid zones.`,
+  minimum_server_version: "4.2.0",
+  maximum_server_version: "",
+  interval_in_seconds: 3600,
```
Comment (trel) [inline src/data/checkfiles/default_checkfiles/validate-user-zone.js:11]: daily/86400?

**Review by trel — COMMENTED** [review]
seems good.

consider pulling out the 999999999 to a well-named variable and reusing it.

**Comment by ganning127** [comment]
> So, is this addressed in this PR? Or we'll work on that separately?

The issue with the alias `findUsersWithInvalidZones` isn't "fixed" in this PR. The issue with `findUsersWithInvalidZones` is a problem with the CPP API, so we'll probably need to work on that separately. However, we're using the alias `usersInvalidZone`, which works great. 

> Should merging this wait

I don't think merging this PR needs to wait for the CPP API to work with `findUsersWithInvalidZones`, as the alias we're using now, `usersInvalidZone` works great and essentially "bypasses" that issue. Open to waiting if there's a good reason though!

**Comment by ganning127** [comment]
Yep, that's what I think it's the particular alias we use. I'm finding that really weird as well... will fix your changes in the above comments soon

**Comment by trel** [comment]
squash and # please

**Comment by ganning127** [comment]
squashed and added #

**Pull request merged**